File: aoc2018/day1/puzzle2.py
results = set()
result = 0
end = False
while not end:
    input = open("input", "r")
    for line in input:
        if result in results:
            end = True
            break
        results.add(result)
        line = line.strip("\n")
        result += int(line)
print(result)

File: aoc2020/day1/puzzle2.pl
:- use_module(library(pio)).
:- use_module(library(clpfd)).
:- set_prolog_flag(double_quotes,codes).

:- dynamic entry/1.

all_chars([]) --> [].
all_chars([Head|Tail]) --> [Head], all_chars(Tail).
parse([]) --> [].
parse([Line|Tail]) --> all_chars(Line), "\n", !, parse(Tail).

assertData([]).
assertData([String|Tail]) :-
     number_codes(Number, String),
     %writeln(Ii),
     assertz(entry(Number)),
     assertData(Tail).

main :-
    phrase_from_file(parse(Data), "input"),
    assertData(Data),
    X + Y + Z #= 2020,
    entry(X), entry(Y), entry(Z),
    Result #= X * Y * Z,
    writeln(Result).
